kernel/src/sched.rs

```diff
@@ -363,6 +363,11 @@ impl Kernel {
                         Some(ContextSwitchReason::SyscallFired { syscall }) => {
                             process.debug_syscall_called(syscall);
 
+                            if let Err(_) = platform.filter_syscall(process, &syscall) {
+                                process.set_syscall_return_value(ReturnCode::ENODEVICE.into());
```

Comment:
I think @bradjc's comment (quoted below) is regarding this.

> Is it a security issue to tell an application that its syscall failed because it isn't permitted? If not, I think this is worthy of its own return code.

My intuition would have also been to pretend like the device doesn't exist. If an application is written to be able to handle working on a board without a particular device, it can also operate without permission to access the syscalls for the device.

Reply:
Hrm.. on second though, I suppose that logic is _only_ true if the policy is to always prevent all access to a particular driver number. But not when the policies only prevents certain systems to a driver, in certain situations (e.g. only to certain commands).

In the latter case, ENODEVICE is kind of the wrong error...